Library/Formula/eralchemy.rb

```diff
@@ -0,0 +1,38 @@
+class Eralchemy < Formula
+  desc "Simple entity relation (ER) diagrams generation"
+  homepage "https://github.com/Alexis-benoist/eralchemy"
+  url "https://pypi.python.org/packages/source/E/ERAlchemy/ERAlchemy-1.0.1.tar.gz"
+  sha256 "98425b21fe0f12ff33e72789c8a308baa1c2d5fdf157388162deaa510551da4f"
+
+  depends_on :python if MacOS.version <= :snow_leopard
+  depends_on "graphviz"
+  depends_on "pkg-config" => :build
+
+  resource "pygraphviz" do
+    url "https://pypi.python.org/packages/source/p/pygraphviz/pygraphviz-1.3.1.tar.gz"
+    sha256 "7c294cbc9d88946be671cc0d8602aac176d8c56695c0a7d871eadea75a958408"
+  end
+
+  resource "SQLAlchemy" do
+    url "https://pypi.python.org/packages/source/S/SQLAlchemy/SQLAlchemy-1.0.11.tar.gz"
+    sha256 "0b24729787fa1455009770880ea32b1fa5554e75170763b1aef8b1eb470de8a3"
+  end
+
+  def install
+    ENV.prepend_create_path "PYTHONPATH", libexec/"vendor/lib/python2.7/site-packages"
+    %w[pygraphviz SQLAlchemy].each do |r|
+      resource(r).stage do
+        system "python", *Language::Python.setup_install_args(libexec/"vendor")
+      end
+    end
+
+    ENV.prepend_create_path "PYTHONPATH", libexec/"lib/python2.7/site-packages"
+    system "python", *Language::Python.setup_install_args(libexec)
+
+    bin.install Dir[libexec/"bin/*"]
+    bin.env_script_all_files(libexec/"bin", :PYTHONPATH => ENV["PYTHONPATH"])
+  end
+  test do
+    system "#{bin}/eralchemy", "-v"
```

Comment:
The test I want to write is the following:
1) Check version
2) Create a test SQLite database
3) Draw the ER diagram of this DB
4) Check that the diagram exists.

For 2), I don't know if I can call this [
https://github.com/Alexis-benoist/eralchemy/blob/master/tests/create_db.py](script)
directly (which will be packaged as a ressource), or if I need to include
it to the eralchemy cli.

2015-12-31 17:23 GMT+01:00 Baptiste Fontaine <notifications@github.com>:

> In Library/Formula/eralchemy.rb
> <https://github.com/Homebrew/homebrew/pull/47515#discussion_r48661864>:
>
> > +  def install
> > +    ENV.prepend_create_path "PYTHONPATH", libexec/"vendor/lib/python2.7/site-packages"
> > +    %w[pygraphviz SQLAlchemy].each do |r|
> > +      resource(r).stage do
> > +        system "python", *Language::Python.setup_install_args(libexec/"vendor")
> > +      end
> > +    end
> > +
> > +    ENV.prepend_create_path "PYTHONPATH", libexec/"lib/python2.7/site-packages"
> > +    system "python", *Language::Python.setup_install_args(libexec)
> > +
> > +    bin.install Dir[libexec/"bin/*"]
> > +    bin.env_script_all_files(libexec/"bin", :PYTHONPATH => ENV["PYTHONPATH"])
> > +  end
> > +  test do
> > +    system "#{bin}/eralchemy", "-v"
>
> Can’t we write a test that does something more substantial than e.g. -v/-h
> ?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Homebrew/homebrew/pull/47515/files#r48661864>.
>